src/system/irrigation.h

```diff
@@ -80,23 +80,31 @@ class SystemTimeProvider : public ITimeProvider {
 
     bool update() {
         bool anySucess = false;
+        DateTime now(getBuildTimeAsUnixTime(__DATE__, __TIME__));
+        DateTime prev = datetime.toDateTime();
         _for_each(providers, _tp, ITimeProvider *)
```

Comment:
I think the priority was to use the NTP, if it fails then use RTC, if NTP works , then need to update RTC, not sure if this logic is doing that

Reply:
It takes the latest hour as the right one, and then updates the other providers with that hour.